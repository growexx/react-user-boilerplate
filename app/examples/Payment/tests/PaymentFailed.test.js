import React from 'react';
import { render, wait } from 'react-testing-library';
import request from 'utils/request';
import PropTypes from 'prop-types';
import { createMemoryHistory } from 'history';
import { BrowserRouter as Router } from 'react-router-dom';
import PaymentFailed from '../PaymentFailed';

const props = {};
jest.mock('utils/request');

const componentWrapper = updatedProps =>
  render(<PaymentFailed {...props} {...updatedProps} />);

function renderWithRouter(
  ui,
  { route = '/', history = createMemoryHistory({ initialEntries: [route] }) },
) {
  const Wrapper = ({ children }) => (
    <Router history={history}>{children}</Router>
  );
  Wrapper.propTypes = {
    children: PropTypes.any,
  };
  return {
    ...render(ui, { wrapper: Wrapper }),
    history,
  };
}
describe('<PaymentFailed />', () => {
  it('should render with no props', () => {
    const {
      container: { firstChild },
    } = componentWrapper();
    expect(firstChild).toMatchSnapshot();
  });

  it('should redner when loading false', async () => {
    const {
      container: { firstChild },
    } = componentWrapper();
    request.mockImplementation(() =>
      Promise.resolve({
        status: 1,
        data: 'data recieved',
        message: 'Success',
      }),
    );

    await wait(() => {
      expect(firstChild).toMatchSnapshot();
    });
  });

  test('payment success redirect', async () => {
    const {
      container: { firstChild },
    } = renderWithRouter(<PaymentFailed />, {
      route:
        '/gateway/success?paymentId=PAYID-MDVHWHY88Y576948Y3372059&token=EC-72S425174G792992P&PayerID=YJT84LQ2W7GU8',
    });
    expect(firstChild).toMatchSnapshot();
  });
});

describe('<PaymentFailed /> reject', () => {
  it('should redner when promise reject', async () => {
    const {
      container: { firstChild },
    } = componentWrapper();
    request.mockImplementation(() =>
      // eslint-disable-next-line prefer-promise-reject-errors
      Promise.reject({
        status: 0,
        error: 'sometihng went wrong',
      }),
    );

    await wait(() => {
      expect(firstChild).toMatchSnapshot();
    });
  });
});
