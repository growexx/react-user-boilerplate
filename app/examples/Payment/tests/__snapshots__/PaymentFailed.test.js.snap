// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentFailed /> payment success redirect 1`] = `
.c0 .common-payment-main {
  width: 60%;
}

.c0 .continue-btn {
  height: 42px;
  margin-left: 10%;
}

.c0 .payment-main-card {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 h2 {
  height: 42px;
  width: 100%;
  margin-bottom: 30px;
  paddind: 20px;
}

.c0 .logo {
  margin-top: 1.5rem;
}

.c0 .pay_btn {
  margin-top: 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .paytm_logo {
  height: 100px;
}

.c0 .suceess-container {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .suceess-container .sub {
  padding: 1rem 0px;
}

.c0 .suceess-container h1 {
  color: #52c41a;
  padding-left: 1rem;
}

.c0 .success-icon-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .success-icon {
  font-size: 3rem;
}

<div
  class="c0"
>
  <div
    class="suceess-container"
  >
    <div
      class="sub"
    >
      <div
        class="success-icon-container"
      >
        <span
          aria-label="close-circle"
          class="anticon anticon-close-circle success-icon"
          role="img"
          style="color: red;"
        >
          <svg
            aria-hidden="true"
            data-icon="close-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M685.4 354.8c0-4.4-3.6-8-8-8l-66 .3L512 465.6l-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155L340.5 670a8.32 8.32 0 00-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3L512 564.4l99.3 118.4 66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.5 515l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
            />
            <path
              d="M512 65C264.6 65 64 265.6 64 513s200.6 448 448 448 448-200.6 448-448S759.4 65 512 65zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
          </svg>
        </span>
        <h1
          style="color: red;"
        >
          Payment Failed!
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`<PaymentFailed /> reject should redner when promise reject 1`] = `
.c0 .common-payment-main {
  width: 60%;
}

.c0 .continue-btn {
  height: 42px;
  margin-left: 10%;
}

.c0 .payment-main-card {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 h2 {
  height: 42px;
  width: 100%;
  margin-bottom: 30px;
  paddind: 20px;
}

.c0 .logo {
  margin-top: 1.5rem;
}

.c0 .pay_btn {
  margin-top: 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .paytm_logo {
  height: 100px;
}

.c0 .suceess-container {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .suceess-container .sub {
  padding: 1rem 0px;
}

.c0 .suceess-container h1 {
  color: #52c41a;
  padding-left: 1rem;
}

.c0 .success-icon-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .success-icon {
  font-size: 3rem;
}

<div
  class="c0"
>
  <div
    class="suceess-container"
  >
    <div
      class="sub"
    >
      <div
        class="success-icon-container"
      >
        <span
          aria-label="close-circle"
          class="anticon anticon-close-circle success-icon"
          role="img"
          style="color: red;"
        >
          <svg
            aria-hidden="true"
            data-icon="close-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M685.4 354.8c0-4.4-3.6-8-8-8l-66 .3L512 465.6l-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155L340.5 670a8.32 8.32 0 00-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3L512 564.4l99.3 118.4 66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.5 515l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
            />
            <path
              d="M512 65C264.6 65 64 265.6 64 513s200.6 448 448 448 448-200.6 448-448S759.4 65 512 65zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
          </svg>
        </span>
        <h1
          style="color: red;"
        >
          Payment Failed!
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`<PaymentFailed /> should redner when loading false 1`] = `
.c0 .common-payment-main {
  width: 60%;
}

.c0 .continue-btn {
  height: 42px;
  margin-left: 10%;
}

.c0 .payment-main-card {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 h2 {
  height: 42px;
  width: 100%;
  margin-bottom: 30px;
  paddind: 20px;
}

.c0 .logo {
  margin-top: 1.5rem;
}

.c0 .pay_btn {
  margin-top: 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .paytm_logo {
  height: 100px;
}

.c0 .suceess-container {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .suceess-container .sub {
  padding: 1rem 0px;
}

.c0 .suceess-container h1 {
  color: #52c41a;
  padding-left: 1rem;
}

.c0 .success-icon-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .success-icon {
  font-size: 3rem;
}

<div
  class="c0"
>
  <div
    class="suceess-container"
  >
    <div
      class="sub"
    >
      <div
        class="success-icon-container"
      >
        <span
          aria-label="close-circle"
          class="anticon anticon-close-circle success-icon"
          role="img"
          style="color: red;"
        >
          <svg
            aria-hidden="true"
            data-icon="close-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M685.4 354.8c0-4.4-3.6-8-8-8l-66 .3L512 465.6l-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155L340.5 670a8.32 8.32 0 00-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3L512 564.4l99.3 118.4 66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.5 515l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
            />
            <path
              d="M512 65C264.6 65 64 265.6 64 513s200.6 448 448 448 448-200.6 448-448S759.4 65 512 65zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
          </svg>
        </span>
        <h1
          style="color: red;"
        >
          Payment Failed!
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`<PaymentFailed /> should render with no props 1`] = `
.c0 .common-payment-main {
  width: 60%;
}

.c0 .continue-btn {
  height: 42px;
  margin-left: 10%;
}

.c0 .payment-main-card {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 h2 {
  height: 42px;
  width: 100%;
  margin-bottom: 30px;
  paddind: 20px;
}

.c0 .logo {
  margin-top: 1.5rem;
}

.c0 .pay_btn {
  margin-top: 1.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .paytm_logo {
  height: 100px;
}

.c0 .suceess-container {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .suceess-container .sub {
  padding: 1rem 0px;
}

.c0 .suceess-container h1 {
  color: #52c41a;
  padding-left: 1rem;
}

.c0 .success-icon-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .success-icon {
  font-size: 3rem;
}

<div
  class="c0"
>
  <div
    class="suceess-container"
  >
    <div
      class="sub"
    >
      <div
        class="success-icon-container"
      >
        <span
          aria-label="close-circle"
          class="anticon anticon-close-circle success-icon"
          role="img"
          style="color: red;"
        >
          <svg
            aria-hidden="true"
            data-icon="close-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M685.4 354.8c0-4.4-3.6-8-8-8l-66 .3L512 465.6l-99.3-118.4-66.1-.3c-4.4 0-8 3.5-8 8 0 1.9.7 3.7 1.9 5.2l130.1 155L340.5 670a8.32 8.32 0 00-1.9 5.2c0 4.4 3.6 8 8 8l66.1-.3L512 564.4l99.3 118.4 66 .3c4.4 0 8-3.5 8-8 0-1.9-.7-3.7-1.9-5.2L553.5 515l130.1-155c1.2-1.4 1.8-3.3 1.8-5.2z"
            />
            <path
              d="M512 65C264.6 65 64 265.6 64 513s200.6 448 448 448 448-200.6 448-448S759.4 65 512 65zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
            />
          </svg>
        </span>
        <h1
          style="color: red;"
        >
          Payment Failed!
        </h1>
      </div>
    </div>
  </div>
</div>
`;
